ipv6: don't disable interface if last ipv6 address is removed

Backport of upstream commit:

commit 876fd05ddbae03166e7037fca957b55bb3be6594
Author: Hannes Frederic Sowa <hannes@stressinduktion.org>
Date:   Mon Jun 24 00:22:20 2013 +0200

    ipv6: don't disable interface if last ipv6 address is removed

diff --git a/net/ipv6/addrconf.c b/net/ipv6/addrconf.c
index 96b5a74..57375fd 100644
--- a/net/ipv6/addrconf.c
+++ b/net/ipv6/addrconf.c
@@ -2290,12 +2290,6 @@ static int inet6_addr_del(struct net *net, int ifindex, const struct in6_addr *p
 			read_unlock_bh(&idev->lock);
 
 			ipv6_del_addr(ifp);
-
-			/* If the last address is deleted administratively,
-			   disable IPv6 on this interface.
-			 */
-			if (list_empty(&idev->addr_list))
-				addrconf_ifdown(idev->dev, 1);
 			return 0;
 		}
 	}
